### PR TITLE
[NUI] Add Tizen.NUI.XamlBuild module

### PR DESCRIPTION
**Pull request by elishateng**
### Description of Change ###

1. Add Tizen.NUI.XamlBuild module. XamlBuild is for XamlG/XamlC feature. It won't be installed to target, and it only works when building in VisualStudio. So no sample apps for it.
2. Fix the CreateDPObject enum mismatch issue.

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by dongsug-song** [comment]
@WonyoungChoi 
안녕하세요. 최원영님,
SRCN 에서 NUI.XamlBuild (https://github.com/nui-dali/Tizen.NUI.XamlBuild) 의 내용을 옮긴 PR 입니다.
dotnet build 관련해서 지식이 부족하고, GBS 빌드로 확인 테스트를 어떻게 해야 하는지 잘 몰라서 리뷰를 요청드립니다.
이 PR 이 GBS 빌드 에러 문제를 해결 할 수 있는지 리뷰 부탁드립니다.
감사합니다.

**Comment by dongsug-song** [comment]
@WonyoungChoi 
안녕하세요.
제가 local 에서 App 이 이 PR 이 반영된 local 빌드된 tizenfx 의 버젼(10.0.0.99999) 에 맞게 수정하고, GBS 빌드를 하니 아래와 같이 성공 하였습니다.  

[  135s] /home/abuild/rpmbuild/BUILD/com.samsung.tv.arkwheel-1.0.0/TextResource/TextResource.csproj : warning NU1603: TextResource depends on Tizen.NET (>= 9.0.0.16562) but Tizen.NET 9.0.0.16562 was not found. An approximate best match of Tizen.NET 10.0.0.99999 was resolved.
...
info: *** Build Status Summary ***
=== Total succeeded built packages: (1) ===
info: generated html format report:
     /home/ssong2best/GBS_BUILD_2022/tizen65-pontus-20220222/local/repos/tizen/armv7l/index.html
info: generated RPM packages can be found from local repo:
     /home/ssong2best/GBS_BUILD_2022/tizen65-pontus-20220222/local/repos/tizen/armv7l/RPMS
...

참고 부탁드립니다. 감사합니다.

**Review by zg2nets — APPROVED** [review]
looks good。

**Review by huayongxu — APPROVED** [review]
lgtm

**Review by zhouleonlei — APPROVED** [review]
LGTM

**Review by AchoWang — APPROVED** [review]
Looks good to me.

**Review by AdunFang — APPROVED** [review]
It is OK.

File: pkg/Tizen.NET.API10/build/Tizen.NET.API10.targets [hunk 1]
```diff
@@ -9,12 +9,15 @@
...
       <Visible>false</Visible>
       <NuGetPackageId>Microsoft.NETCore.App.Ref</NuGetPackageId>
       <NuGetPackageVersion>$(_TizenNETCoreAppPackageVersion)</NuGetPackageVersion>
     </Reference>
   </ItemGroup>
 
+  <ImportGroup Condition=" '$(ExcludeRestorePackageImports)' != 'true' ">
+    <Import Project="$(MSBuildThisFileDirectory)..\buildTransitive\Tizen.NUI.XamlBuild.targets" Condition="Exists('$(MSBuildThisFileDirectory)..\buildTransitive\Tizen.NUI.XamlBuild.targets')" />
+  </ImportGroup>
 </Project>
```
Comment (dongsug-song) [inline pkg/Tizen.NET.API10/build/Tizen.NET.API10.targets:23]: this is trivial thing but could you add new line at the end of file?
- reference: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Comment by dongsug-song** [comment]
@elishateng 
could you add more detail in description?
the purpose of this XamlBuild pkg and please mention that this is not used and not installed in target devices.

**Comment by dongsug-song** [comment]
@elishateng 
by this change, the `<NeedInjection>False</NeedInjection>` in csproj, still work?
I mean, true(xaml_g/c) and false(examl) config is properly working?

File: src/Tizen.NUI.XamlBuild/src/internal/Xaml/DesignMode.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Tizen.NUI.Xaml
```
Comment (dongsug-song) [inline src/Tizen.NUI.XamlBuild/src/internal/Xaml/DesignMode.cs:1]: please add copy right statements

File: src/Tizen.NUI.XamlBuild/src/internal/Xaml/ExpandMarkupsVisitor.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                    //name = split [0];
+                }
+
+                Type type;
+                var typeResolver = serviceProvider.GetService(typeof (IXamlTypeResolver)) as IXamlTypeResolver;
+                if (typeResolver == null)
+                    type = null;
+                // Add Binding and StaticResource support, The ordinal code can't find BindingExtension for Binding
+                //else if (match == "Binding")
+                //{
```
Comment (dongsug-song) [inline src/Tizen.NUI.XamlBuild/src/internal/Xaml/ExpandMarkupsVisitor.cs:139]: please remove useless lines `//`

File: src/Tizen.NUI.XamlBuild/src/internal/Xaml/MarkupExpressionParser.cs [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+//
+//
+// Author(s):
+//   Moonlight List (moonlight-list@lists.ximian.com)
+//   Stephane Delcroix (stephane@mi8.be)
+//
+// Copyright 2009 Novell, Inc.
+// Copyright 2013 Xamarin, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
```
Comment (dongsug-song) [inline src/Tizen.NUI.XamlBuild/src/internal/Xaml/MarkupExpressionParser.cs:14]: please check the copyright issue.
is it no problem to copy this file and to use it?
also please check other files in this PR.

**Comment by dongsug-song** [comment]
@AdunFang @elishateng 
could you add some sample App or test cases to verify this pkg?

**Comment by elishateng** [comment]
> 

Yes, it still works.

**Comment by elishateng** [comment]
> @elishateng by this change, the `<NeedInjection>False</NeedInjection>` in csproj, still work? I mean, true(xaml_g/c) and false(examl) config is properly working?

Yes, it still works.

**Review by dongsug-song — APPROVED** [review]
looks good to me

**Review by Seoyeon2Kim — APPROVED** [review]

**Review by JoogabYun — APPROVED** [review]

**Pull request merged**